### DISCUSSION TITLE
two "register" on the same field

### DISCUSSION THREAD

**Discussion by ranran2121** [react-hook-form/react-hook-form#9656] [Q&A]
Hi, 
i would like to know if I can register the same input twice, and have something like to following snippet to work:

```
 <input
                  type="checkbox"
                  value={value}
                  {...register(value, {
                    required: false,
                  })}
                {...register("values", {
                     validate: () => getValues("values").length >= 1,
                  })}
                />
```
the reason is that for the validation I need to group some inputs and throw an error only if the user has not selected any of them. On the other hand, it is easier also to have single values in order to set default values and retrieve them after submission rather than doing login on the array

**Comment by bluebill1049** [comment]
can you write the logic into the single `validate` function?

**Reply by ranran2121** [reply]
I can write the logic if I use the collective field (i.e. "values") since I can check its length, but I cannot do the same on the individual (i.e. value) since that field does not know about the others
